src/runtime_src/core/pcie/driver/linux/xocl/xocl_subdev.c

```diff
@@ -561,10 +602,14 @@ static int __xocl_subdev_create(xdev_handle_t xdev_hdl,
 	struct resource *res = NULL;
 	int i, retval;
 	uint32_t dev_idx = 0;
+	bool is_reserved = false;
 
-	retval = xocl_subdev_reserve(xdev_hdl, sdev_info, &subdev);
-	if (retval)
-		goto error;
+	is_reserved = xocl_subdev_check_reserve(xdev_hdl, sdev_info, &subdev);
+	if (!is_reserved) {
+		retval = __xocl_subdev_reserve(xdev_hdl, sdev_info, &subdev);
+		if (retval)
+			goto error;
+	}
```

Comment:
How about changing 607- 612 to something like:
retval = __xocl_subdev_reserve(xdev_hdl, sdev_info, &subdev);
if ( retval != -EXIST || subdev->state != XOCL_SUBDEV_STATE_INIT) {
        /* subdev is not reserved */
        goto error;
}
And remove xocl_subdev_check_reserve().

Reply:
Actually, __xocl_subdev_reserve function fails if (subdev->state != XOCL_SUBDEV_STATE_UNINIT)
Hence, this will not work. We may need to change the implementation of __xocl_subdev_reserve  function. 

Actually __xocl_subdev_reserve  will be called two times. One at the time of deciding the subdevice index and another time to create subdevice. Hence we can't make it this way, I believe.
This way we may need to change the code more.